core/sql/generator/GenExpGenerator.cpp

```diff
@@ -753,12 +752,42 @@ void ExpGenerator::copyDefaultValues(
 
       if (srcAttr->getDefaultValue())
 	{
-	  char * tgtDefVal =
-	    new(generator->getSpace()) char[srcAttr->getDefaultValueStorageLength()];
-	  
-	  str_cpy_all(tgtDefVal, 
-		      srcAttr->getDefaultValue(), 
-		      srcAttr->getDefaultValueStorageLength());
+          Lng32 tgtDefLen = tgtAttr->getDefaultValueStorageLength();
+          Lng32 srcDefLen = srcAttr->getDefaultValueStorageLength();
+          char* srcDefVal = srcAttr->getDefaultValue();
+	  char * tgtDefVal = new(generator->getSpace()) char[tgtDefLen];
+          // if source and target def storage lengths dont match, then
+          // need to move each part (null, vclen, data) separately.
+          if (tgtDefLen != srcDefLen) 
```

Comment:
Might we be fooled into thinking the tgtDefLen is equal to the srcDefLen, but src is nullable varchar(n-2) while tgt is non-nullable varchar(n)?